### PR TITLE
Docs: Fix typos in documentation files

### PR DESCRIPTION
**Pull request by ember91**
This includes:
 - .rst
 - .md
 - .txt
 - Misc/HISTORY
 - Tool/freeze/README

See https://github.com/python/cpython/pull/125473 for context.

----
📚 Documentation preview 📚: https://cpython-previews--125569.org.readthedocs.build/

### CODE REVIEW

**Comment by ember91** [comment]
I Have included changes to `MISC/HISTORY` as well as the `Misc/News.d/*.rst` files here just to show them. If you think they shouldn't be included because they modify history, just tell me.

I will follow up with more PRs.

File: Lib/test/test_doctest/test_doctest2.txt [hunk 1]
```diff
@@ -10,5 +10,5 @@ This test also has some (random) encoded (utf-8) unicode text:
 
   ï»¿ÐÐÐÐÐ
 
-This doesn't cause a problem in the tect surrounding the examples, but
+This doesn't cause a problem in the text surrounding the examples, but
```
Comment (ember91) [inline Lib/test/test_doctest/test_doctest2.txt]: It could be `test` as well but I think it's `text`.

**Comment by rruuaanng** [comment]
Recommend add a **Docs** title label, like this
```
Docs: Fix typos in documentation files
```

File: Misc/HISTORY [hunk 2]
```diff
@@ -168,7 +168,7 @@ Core and Builtins
 - Issue #24806: Prevent builtin types that are not allowed to be subclassed from
   being subclassed through multiple inheritance.
 
-- Issue #24848: Fixed a number of bugs in UTF-7 decoding of misformed data.
+- Issue #24848: Fixed a number of bugs in UTF-7 decoding of malformed data.
```
Comment (skirpichev) [inline Misc/HISTORY]: I think this should be reverted.  It match commit message.

**Review by skirpichev — COMMENTED** [review]
LGTM with one nitpick

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -440,7 +440,7 @@ jobs:
     with:
       config_hash: ${{ needs.check_source.outputs.config_hash }}
       options: ./configure --config-cache --with-thread-sanitizer --with-pydebug
-      suppressions_path: Tools/tsan/supressions.txt
+      suppressions_path: Tools/tsan/suppressions.txt
```
Comment (skirpichev) [inline .github/workflows/build.yml]: This is just invalid.  Please check path.

Please also avoid force-pushes.

**Comment by skirpichev** [comment]
IMO, it already touch enough files.

File: Misc/NEWS.d/3.5.3rc1.rst [hunk 4]
```diff
@@ -1984,7 +1984,7 @@ Haubenwallner.
 .. nonce: CLz6qy
 .. section: Build
 
-Rename --with-optimiations to --enable-optimizations.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/3.5.3rc1.rst:1987]: Did you check the Git log to see if this renaming was an actual configure flag spelling bugfix? Ditto for the 3.6 and 3.7 NEWS rst files.

File: Misc/HISTORY (hunk 2 above)
Comment (ember91) [reply Misc/HISTORY]: Done!

File: .github/workflows/build.yml (hunk 3 above)
Comment (ember91) [reply .github/workflows/build.yml]: Yes this change was from a messed up push.

**Comment by ember91** [comment]
> IMO, it already touch enough files.

Yes I messed up. Have a look again. It should be similar to before.

File: Misc/NEWS.d/3.5.3rc1.rst (hunk 4 above)
Comment (willingc) [reply Misc/NEWS.d/3.5.3rc1.rst:1987]: I'm wondering if it makes sense at all to go back through old versions and correct typos.

Comment (erlend-aasland) [reply Misc/NEWS.d/3.5.3rc1.rst:1987]: Yeah. There is a risk of _introducing_ documentation errors and inconsistencies with such changes.

Comment (ember91) [reply Misc/NEWS.d/3.5.3rc1.rst:1987]: We can start with the second question here from @willingc. That's my question to you. I included the changes in history files to showcase but I don't know if it should be included.

**Review by willingc — CHANGES_REQUESTED** [review]
@ember91 I see that you are a first time contributor. Welcome!

We discourage large PRs, such as this one with 29 files, that are correcting typos. Large PRs are more time consuming for reviewers. They also run the risk that a typo change in tests, tools, or library code will actually cause breakage. In addition, a PR with many files will notify many core developers unnecessarily. It's better to keep PRs small. :sunny:

I recommend that you split the PR and do the following:
- 1 PR covering changes in the `Doc` directory
- 1 PR covering changes in the `InternalDocs` directory
- Do not open PRs for `Lib`, `Misc`, `Objects`, or `Tools`

Thanks!

File: Misc/NEWS.d/3.5.3rc1.rst (hunk 4 above)
Comment (willingc) [reply Misc/NEWS.d/3.5.3rc1.rst:1987]: @erlend-aasland I think https://github.com/python/cpython/pull/125569#pullrequestreview-2371548312 is how we should approach typo PRs.

**Comment by sobolevn** [comment]
@ember91, welcome to CPython 👋 

I agree with @willingc, please, don't create large PRs since they also spawn notifications for a lot of codeowners for the changed files.

The only thing that I would like to highlight is that changing `Misc/` directory is fine, because it is just for docs.

**Comment by ember91** [comment]
> @ember91, welcome to CPython 👋
> 
> I agree with @willingc, please, don't create large PRs since they also spawn notifications for a lot of codeowners for the changed files.
> 
> The only thing that I would like to highlight is that changing `Misc/` directory is fine, because it is just for docs.

Sorry about that. If you want to I can split this up into one PR per codeowner. Or if you think the damage is done already we can keep it as is.

File: Misc/NEWS.d/3.5.3rc1.rst (hunk 4 above)
Comment (vadmium) [reply Misc/NEWS.d/3.5.3rc1.rst:1987]: FWIW the spelling fix is appropriate. This is the original commit in 3.5: https://hg.python.org/cpython/rev/c0ea81315fb6

File: Lib/test/test_doctest/test_doctest2.txt [hunk 5]
```diff
@@ -10,5 +10,5 @@ This test also has some (random) encoded (utf-8) unicode text:
 
   ï»¿ÐÐÐÐÐ
 
-This doesn't cause a problem in the tect surrounding the examples, but
+This doesn't cause a problem in the test surrounding the examples, but
```
Comment (vadmium) [inline Lib/test/test_doctest/test_doctest2.txt:13]:
```suggestion
This doesn't cause a problem in the text surrounding the examples, but
```

I agree with your previous fix. C is closer to X than S on my keyboard. The examples are the doc-_tests_, embedded in the _text_.

File: Misc/NEWS.d/3.5.3rc1.rst (hunk 4 above)
Comment (erlend-aasland) [reply Misc/NEWS.d/3.5.3rc1.rst:1987]: > FWIW the spelling fix is appropriate. This is the original commit in 3.5: https://hg.python.org/cpython/rev/c0ea81315fb6

I know, and for such cases it is helpful to include this Git archeology in the description of the PR, for the convenience of the reviewer.

**Review by JelleZijlstra — APPROVED** [review]

**Comment by ember91** [comment]
Closing this PR. I will follow the suggestions by @willingc in https://github.com/python/cpython/pull/125569#pullrequestreview-2371548312, but also its follow up https://github.com/python/cpython/pull/125569#issuecomment-2416174012.

I will apply all suggestions in this PR this far in new PRs, obviously.

**Comment by willingc** [comment]
Thanks @ember91 for being receptive to our feedback. Looking forward to your continued contributions.

**Pull request closed without merging**